### PR TITLE
Customize pihole-FTL.conf with FTLCONF_ prefixed environment variables.

### PR DESCRIPTION
**Pull request by AlexandreRatte**
## Description

Parse all environment variables starting with the `FTLCONF_` prefix and setup the variable's suffix as a `pihole-FTL.conf` customized setting. For example, setting the `FTLCONF_REPLY_ADDR4=10.1.2.3` environment variable would result in a `pihole-FTL.conf` setting of `REPLY_ADDR4=10.1.2.3`.

~~This PR is marked as having a breaking change since the yet undocumented `ServerIP` and `ServerIPv6` variables have been deprecated in favor of provisioning them as `FTLCONF_REPLY_ADDR4` and `FTLCONF_REPLY_ADDR6` respectively.~~

## Motivation and Context

Addresses #922 and #919

There were use cases where `pihole-FTL.conf` could be supplied through a read-only filesystem i.e.: a Kubernetes ConfigMap, and calling the `changeFTLsetting` function on such a read-only file would cause the container to crash loop. This PR is meant to provide the pi-hole administrator all the flexibility that was previously associated with volume-mounting a customized `pihole-FTL.conf`, but now by using environment variables.

## How Has This Been Tested?

- `docker run -it -e FTLCONF_REPLY_ADDR4=10.1.2.3 -e FTLCONF_MAXDBDAYS=7 issue922`
- Shelled into the running container then executed:
```
# cat /etc/pihole/pihole-FTL.conf
PRIVACYLEVEL=0
MAXDBDAYS=7
REPLY_ADDR4=10.1.2.3
#
```

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by PromoFaux** [comment]
I've not tested this functionally, yet, but I like the idea a lot. I am wondering (out loud) if we can do something similar for `setupVars.conf` keypairs, so `SETUPVARS_` env variables, and what this would look like. 

I also need to check a few things, but we might be able to keep the `ServerIP` ones around so that this does not become a breaking change (although I'm all for breaking changes if they are good and sensible like this)

**Comment by AlexandreRatte** [comment]
Yes, implementing something similar for `setupVars.conf` would probably clean things up rather nicely. However, in light of #922 causing issues to folks who are supplying `pihole-FTL.conf` through a read-only volume mount due to the lack of customization capabilities by other means, I would suggest attacking `setupVars.conf` outside of this PR.

Let me know of your thoughts, I'm happy to discuss and make amendments per your review.

File: start.sh [hunk 1]
```diff
@@ -76,8 +76,6 @@ prepare_configs
 [ -n "${INSTALL_WEB_SERVER}" ] && change_setting "INSTALL_WEB_SERVER" "$INSTALL_WEB_SERVER"
 [ -n "${INSTALL_WEB_INTERFACE}" ] && change_setting "INSTALL_WEB_INTERFACE" "$INSTALL_WEB_INTERFACE"
 [ -n "${LIGHTTPD_ENABLED}" ] && change_setting "LIGHTTPD_ENABLED" "$LIGHTTPD_ENABLED"
-[ -n "${ServerIP}" ] && changeFTLsetting "REPLY_ADDR4" "$ServerIP"
```
Comment (PromoFaux) [inline start.sh:79]: For ease of updating, I think we should leave these two lines in - but add a note to the README/release notes that they will be removed in a later version, and should be replaced with `FTLCONF_REPLY_ADDR4/6` 

I'll look at untangling `ServerIP` from the rest of the codebase at some point:
image

**Review by PromoFaux — CHANGES_REQUESTED** [review]
For ease of updating, I think we should leave these two lines in - but add a note to the README/release notes that they will be removed in a later version, and should be replaced with `FTLCONF_REPLY_ADDR4/6` 

I'll look at untangling `ServerIP` from the rest of the codebase at some point:
image

File: start.sh (hunk 1 above)
Comment (PromoFaux) [inline start.sh:79]: For ease of updating, I think we should leave these two lines in - but add a note to the README/release notes that they will be removed in a later version, and should be replaced with `FTLCONF_REPLY_ADDR4/6` 

I'll look at untangling `ServerIP` from the rest of the codebase at some point:
image

**Review by PromoFaux — CHANGES_REQUESTED** [review]

File: start.sh (hunk 1 above)
Comment (AlexandreRatte) [reply start.sh:79]: Ok, I've reintroduced parsing of the `ServerIP*` variables and made changes to `README.md` which should highlight their deprecated nature, while also explicitly mentioning that `FTLCONF_REPLY_ADDR4` is recommended.

**Review by PromoFaux — APPROVED** [review]

**Pull request merged**